### DISCUSSION TITLE
[question] Multi service Datadog dashboard setup

### DISCUSSION THREAD

**Discussion by IanVaughan** [zendesk/racecar#302] [Q&A] [answered]
I have a question here on SO, but its not getting any answers
https://stackoverflow.com/questions/73468917/best-way-to-configure-zendesk-racecar-with-many-services

I am using the cool monitoring feature: https://github.com/zendesk/racecar#datadog-monitoring

Which works well with our single service. Now we are expanding into more services, I'm wondering what the correct config for monitoring is:

1. Only have it enabled on one app
    1. Simple, but I believe this will then [miss consumer metrics](https://github.com/zendesk/racecar/blob/0281f590c6e7ed60b8a6b7ee3c22e2d6a641ea67/lib/racecar/datadog.rb#L102)
1. Have it enabled in all apps using the default racecar namespace
    1. This would cause competing/duplicate data for most (all?) metrics
    1. But then nicely pushes all data into the same namespace/dashboard
1. Have it enabled in all apps each with its own namespace (eg setting `datadog_namespace`)
    1. This would duplicate most of the data into different metric namespaces (eating into custom metrics counts)
    1. Also means each service needs a copy of the Racecar dashboard, with all the namespaces changed

**Comment by dasch** [comment, marked as answer]
If you're running your services on Kubernetes then Datadog is likely already scraping some info and using that to populate tags such as `kube_namespace` and whatnot; I recommend leaning on those. You may need to set up that scraping.
